src/server/editorServices.ts

```diff
@@ -1251,7 +1284,10 @@ namespace ts.server {
                     return error;
                 }
                 else {
-                    const oldFileNames = project.compilerService.host.roots.map(info => info.fileName);
+                    // if the project is too large, the root files might not have been all loaded if the total
+                    // program size reached the upper limit. In that case project.projectOptions.files should 
+                    // be more precise. However this would only happen for configured project.
+                    const oldFileNames = project.projectOptions ? project.projectOptions.files : project.compilerService.host.roots.map(info => info.fileName);

```

Comment:
This `files` is the parsed file list for a project, not just what the user specified in `files` list in tsconfig.json. All `projectOptions` are obtained from calling `configFileToProjectOptions`, and in that function [it is guaranteed to have a `files` array](https://github.com/zhengbli/TypeScript/blob/projectSizeLimitFor18-2/src/server/editorServices.ts#L1204). So it should be safe to call `projectOptions.files`